lib/mongo/collection/view/readable.rb

```diff
@@ -254,11 +270,12 @@ def estimated_document_count(opts = {})
           end
 
           %i[limit skip].each do |opt|
-            if @options.key?(opt)
+            if options.key?(opt) || opts.key?(opt)
               raise ArgumentError, "Cannot call estimated_document_count when querying with #{opt}"
```

Comment:
This would cause existing use of `estimated_document_count` to fail when view has skip/limit applied where it worked previously with skip/limit being ignored, right?

Reply:
Previously skip/limit would only cause a failure when it was in the view options

After this change it will also failed when being passed into the method parameters.